packages/core/deprecations/core-deprecations-server-internal/src/deprecations/i18n_texts.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { RouterDeprecatedRouteDetails } from '@kbn/core-http-server';
+import { CoreDeprecatedApiUsageStats } from '@kbn/core-usage-data-server';
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+
+export const getApiDeprecationTitle = (details: RouterDeprecatedRouteDetails) => {
+  const { routePath, routeMethod } = details;
+  const routeWithMethod = `${routeMethod.toUpperCase()} ${routePath}`;
+  const deprecationTypeText = getApiDeprecationTypeText(details);
+
+  return i18n.translate('core.deprecations.deprecations.apiDeprecationInfoTitle', {
+    defaultMessage: 'The "{routeWithMethod}" route {deprecationTypeText}',
+    values: {
+      routeWithMethod,
+      deprecationTypeText,
+    },
+  });
+};
+
+export const getApiDeprecationTypeText = (details: RouterDeprecatedRouteDetails) => {
+  const { routeDeprecationOptions } = details;
+  const deprecationType = routeDeprecationOptions.reason.type;
+
+  return i18n.translate('core.deprecations.deprecations.apiDeprecationType', {
+    defaultMessage:
+      '{deprecationType, select, remove {will be removed} bump {has a new version bump} migrate {has been migrated to a different API} other {has been marked as deprecated}}',
```

Comment:
it `will be xx` since the deprecations are preperations for future actions. the actual APIs are still there

Reply:
You mean they are still there after the upgrade and will be removed in a later version? If so, feel free to ignore my suggestion.

I understood this as follows:
- I'm a user upgrading from an older version (for ex 8.15) to a newer version (for ex 8.16)
- The UA informs me about what is changing in that newer version (hence calling for past or present tense rather than future)
     - In 8.16, this API has moved to that other API that you must use instead.
     - In 8.16, this API no longer exists.
     - In 8.16, this API is deprecated. It will be removed in a future version.
     - In 8.16, this API has a newer version that you should use instead.